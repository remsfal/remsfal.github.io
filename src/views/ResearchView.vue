<script setup lang="ts">
import Section from '@/components/Section.vue'
import ThesisSection from '@/components/ThesisSection.vue'
import TitleWidget from '@/components/TitleWidget.vue'
</script>

<template>
  <TitleWidget title="Forschung" />

  <Section title="Forschung an der HTW Berlin">
    <img alt="HTW Berlin logo" src="@/assets/htw-berlin-logo.svg" class="float-right p-2 htw-logo" />
    <p>
      REMSFAL wurde 2023 von
      <a href="https://www.htw-berlin.de/hochschule/personen/person/?eid=13708"
         class="underline hover:decoration-green-600" target="_blank">
        Prof. Dr.-Ing. Alexander Stanik
      </a>
      als interdisziplinäres wissenschaftliches
      Open Source Forschungsprojekt gegründet und konzentriert sich auf die Entwicklung skalierbarer Architekturen
      und Technologien für verteilte und parallele Informationssysteme in der Immobilienwirtschaft.
      Ziel des Projektes ist die Digitalisierung von Hausverwaltungen durch innovative Verwaltungs-
      und Immobiliensoftware voranzutreiben. Ein besonderer Fokus liegt dabei auf der Optimierung
      von Mängelmanagement-Prozessen und der Implementierung nachhaltiger Strategien zur Reduzierung
      von CO<sub>2</sub>-Emissionen in der Gebäudebewirtschaftung.
      Durch die Erforschung und Erprobung neuer Technologien sollen Prozesse zur Liegenschaftsverwaltung
      effizienter gestaltet und gleichzeitig ökologische Aspekte berücksichtigt werden.
    </p>
    <p>
      Das REMSFAL Projekt ist mehr als eine Open Source Initiative und konzentriert sich unter der Leitung von
      <a href="https://www.htw-berlin.de/hochschule/personen/person/?eid=13708"
         class="underline hover:decoration-green-600" target="_blank">
        Prof. Dr.-Ing. Alexander Stanik von der Hochschule für Technik und Wirtschaft (HTW) Berlin
      </a>
      ebenso auf den Einsatz in der Forschung und der Lehre.
      Studierende erhalten mit diesem Projekt die Möglichkeit
      eigene kreative Ideen mit innovativen Technologien und hohem Praxisbezug
      sowie einer Vielfalt an Anwendungskomplexitäten der Wirtschaftsinformatik
      zu realisieren. Mit dem Themenschwerpunkt auf Entwicklung und Erprobung neuer Technologien und Anwendungen
      für nachhaltige und klimagerechte Gebäudebewirtschaftung ist dieses Projekt seit 2024 auch Teil
      des Forschungsclusters
      <a href="https://www.htw-berlin.de/forschung/forschungsprofil/forschungscluster/cluster/?eid=23"
         class="underline hover:decoration-green-600" target="_blank">
        Klimagerechte Energieversorgungssysteme und energieeffiziente Gebäude (KEG)
      </a>
      der HTW Berlin.
    </p>
  </Section>

  <Section title="Abschlussarbeiten im Studiengang Wirtschaftsinformatik">
    <ThesisSection />
  </Section>
</template>

<style scoped>
.htw-logo {
  height: 230px;
  margin-left: 5px;
  margin-right: 50px;
  margin-bottom: 3px;
}
@media screen and (max-width: 1024px) {
  .htw-logo {
    margin-left: 1px;
  }
}
</style>
