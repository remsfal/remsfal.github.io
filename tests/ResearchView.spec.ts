import { mount, VueWrapper } from '@vue/test-utils'
import { describe, it, expect, beforeEach } from 'vitest'
import ResearchView from '@/views/ResearchView.vue'

// Mock fetchIssues to return dummy data
vi.mock('@/services/GitHubService.ts', async () => {
  const actual = await vi.importActual<any>('@/services/GitHubService.ts')
  return {
    ...actual,
    default: vi.fn(async () => [
      {
        id: 1,
        title: 'First issue',
        body: '**Bold text** with _italic_',
        state: 'open',
        labels: [{ id: 10, name: 'bug', color: 'f00' }],
        html_url: 'https://github.com/example/issue1'
      },
      {
        id: 2,
        title: 'Second issue',
        body: 'Simple issue description',
        state: 'completed',
        labels: [{ id: 11, name: 'enhancement', color: '0f0' }],
        html_url: 'https://github.com/example/issue2'
      }
    ])
  }
})

describe('ResearchView', () => {
  let wrapper: VueWrapper

  beforeEach(async () => {
    wrapper = mount(ResearchView)
    // Wait for fetchIssues to complete
    await new Promise(resolve => setTimeout(resolve, 10))
  })

  

  it('should display issues', () => {
    const issueCards = wrapper.findAll('.issue-card')
    expect(issueCards.length).toBeGreaterThan(0)
  })

  it('should display correct content in issue cards', () => {
    const issueCards = wrapper.findAll('.issue-card')
    for (const card of issueCards) {
      const title = card.find('.issue-title')
      const description = card.find('.issue-description')
      const footer = card.find('.issue-footer')
      expect(title.exists()).toBe(true)
      expect(description.exists()).toBe(true)
      expect(footer.exists()).toBe(true)
    }
  })

  it('should render markdown content correctly', () => {
    const card = wrapper.find('.issue-card')
    const html = card.find('.issue-description').html()
    expect(html).toContain('<strong>Bold text</strong>')
    expect(html).toContain('<em>italic</em>')
  })

  it('should display label badges', () => {
    const labels = wrapper.findAll('.label')
    expect(labels.length).toBeGreaterThan(0)
    expect(labels[0].text()).toBe('bug')
  })

  it('should filter issues based on status', async () => {
    const select = wrapper.find('.filter-select')
    await select.setValue('completed')

    const issueCards = wrapper.findAll('.issue-card')
    expect(issueCards.length).toBe(1)
    expect(issueCards[0].text()).toContain('Second issue')
  })

  it('should filter issues based on search query', async () => {
    const input = wrapper.find('.search-input')
    await input.setValue('first')

    const issueCards = wrapper.findAll('.issue-card')
    expect(issueCards.length).toBe(1)
    expect(issueCards[0].text()).toContain('First issue')
  })

  it('should open correct link for details', () => {
    const link = wrapper.find('.details-link')
    expect(link.attributes('href')).toContain('https://github.com/example/issue')
  })

  it('should show correct status badge text and color', () => {
    const badges = wrapper.findAll('.status-badge')
    expect(badges[0].text()).toBe('Offen')
    expect(badges[1].text()).toBe('Abgeschlossen')
    expect(badges[0].attributes('style')).toContain('background-color')
  })

  it('should log an error when fetchIssues fails', async () => {
    const mockConsoleError = vi.spyOn(console, 'error').mockImplementation(() => {})
  
    // Override the mock to throw
    const errorMessage = 'Network error'
    const fetchIssues = (await import('@/services/GitHubService.ts')).default
    fetchIssues.mockRejectedValueOnce(new Error(errorMessage))
  
    // Remount component
    const wrapper = mount(ResearchView)
    await new Promise(resolve => setTimeout(resolve, 10))
  
    expect(mockConsoleError).toHaveBeenCalledWith('Failed to fetch issues:', expect.any(Error))
  
    mockConsoleError.mockRestore()
  })

  it('should show no issues for unmatched search query', async () => {
    const input = wrapper.find('.search-input')
    await input.setValue('nonexistent issue')
  
    const issueCards = wrapper.findAll('.issue-card')
    expect(issueCards.length).toBe(0)
  })
  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  
  
  
})
