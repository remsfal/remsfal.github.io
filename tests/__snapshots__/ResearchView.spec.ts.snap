// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ResearchView > matches snapshot 1`] = `
"<div data-v-f8d2ac06="" data-v-53ee1ad1="" class="flex flex-col pt-6 px-6 lg:px-20 overflow-hidden title-widget">
  <div data-v-f8d2ac06="" class="mx-6 md:mx-20 mt-0 md:mt-6">
    <h1 data-v-f8d2ac06="" class="text-5xl md:text-6xl font-bold text-gray-900 leading-tight">Forschung</h1>
  </div>
</div>
<div data-v-33107760="" data-v-53ee1ad1="" id="legal" class="py-6 px-6 lg:px-20 mx-0 my-12 lg:mx-20">
  <div data-v-33107760="" class="p-card p-component" data-pc-name="card" pc0="" data-pc-section="root">
    <!---->
    <div class="p-card-body" data-pc-section="body">
      <div class="p-card-caption" data-pc-section="caption">
        <div class="p-card-title" data-pc-section="title">
          <div data-v-33107760="" class="section-title text-2xl m-8">Forschung an der HTW Berlin</div>
        </div>
        <!---->
      </div>
      <div class="p-card-content" data-pc-section="content">
        <div data-v-33107760="" class="section-content [&amp;>p]:text-lg [&amp;>p]:m-8 [&amp;>h3]:font-semibold [&amp;>h3]:text-xl [&amp;>h3]:ml-8 [&amp;>h3]:-mb-4 [&amp;>h3:first-child]:mt-6"><img data-v-53ee1ad1="" alt="HTW Berlin logo" src="/src/assets/htw-berlin-logo.svg" class="float-right p-2 htw-logo">
          <p data-v-53ee1ad1=""> REMSFAL wurde 2023 von <a data-v-53ee1ad1="" href="https://www.htw-berlin.de/hochschule/personen/person/?eid=13708" class="underline hover:decoration-green-600" target="_blank"> Prof. Dr.-Ing. Alexander Stanik </a> als interdisziplinäres wissenschaftliches Open Source Forschungsprojekt gegründet und konzentriert sich auf die Entwicklung skalierbarer Architekturen und Technologien für verteilte und parallele Informationssysteme in der Immobilienwirtschaft. Ziel des Projektes ist die Digitalisierung von Hausverwaltungen durch innovative Verwaltungs- und Immobiliensoftware voranzutreiben. Ein besonderer Fokus liegt dabei auf der Optimierung von Mängelmanagement-Prozessen und der Implementierung nachhaltiger Strategien zur Reduzierung von CO<sub data-v-53ee1ad1="">2</sub>-Emissionen in der Gebäudebewirtschaftung. Durch die Erforschung und Erprobung neuer Technologien sollen Prozesse zur Liegenschaftsverwaltung effizienter gestaltet und gleichzeitig ökologische Aspekte berücksichtigt werden. </p>
          <p data-v-53ee1ad1=""> Das REMSFAL Projekt ist mehr als eine Open Source Initiative und konzentriert sich unter der Leitung von <a data-v-53ee1ad1="" href="https://www.htw-berlin.de/hochschule/personen/person/?eid=13708" class="underline hover:decoration-green-600" target="_blank"> Prof. Dr.-Ing. Alexander Stanik von der Hochschule für Technik und Wirtschaft (HTW) Berlin </a> ebenso auf den Einsatz in der Forschung und der Lehre. Studierende erhalten mit diesem Projekt die Möglichkeit eigene kreative Ideen mit innovativen Technologien und hohem Praxisbezug sowie einer Vielfalt an Anwendungskomplexitäten der Wirtschaftsinformatik zu realisieren. Mit dem Themenschwerpunkt auf Entwicklung und Erprobung neuer Technologien und Anwendungen für nachhaltige und klimagerechte Gebäudebewirtschaftung ist dieses Projekt seit 2024 auch Teil des Forschungsclusters <a data-v-53ee1ad1="" href="https://www.htw-berlin.de/forschung/forschungsprofil/forschungscluster/cluster/?eid=23" class="underline hover:decoration-green-600" target="_blank"> Klimagerechte Energieversorgungssysteme und energieeffiziente Gebäude (KEG) </a> der HTW Berlin. </p>
        </div>
      </div>
      <!---->
    </div>
  </div>
</div>
<div data-v-33107760="" data-v-53ee1ad1="" id="legal" class="py-6 px-6 lg:px-20 mx-0 my-12 lg:mx-20">
  <div data-v-33107760="" class="p-card p-component" data-pc-name="card" pc1="" data-pc-section="root">
    <!---->
    <div class="p-card-body" data-pc-section="body">
      <div class="p-card-caption" data-pc-section="caption">
        <div class="p-card-title" data-pc-section="title">
          <div data-v-33107760="" class="section-title text-2xl m-8">Abschlussarbeiten im Studiengang Wirtschaftsinformatik</div>
        </div>
        <!---->
      </div>
      <div class="p-card-content" data-pc-section="content">
        <div data-v-33107760="" class="section-content [&amp;>p]:text-lg [&amp;>p]:m-8 [&amp;>h3]:font-semibold [&amp;>h3]:text-xl [&amp;>h3]:ml-8 [&amp;>h3]:-mb-4 [&amp;>h3:first-child]:mt-6">
          <div data-v-0ce74e49="" data-v-53ee1ad1="" class="issues-container">
            <!-- Controls Section -->
            <div data-v-0ce74e49="" class="controls-section">
              <div data-v-0ce74e49="" class="search-box"><input data-v-0ce74e49="" type="text" placeholder="Suchen Sie nach Projekten..." class="search-input"></div>
              <div data-v-0ce74e49="" class="filter-box"><select data-v-0ce74e49="" class="filter-select">
                  <option data-v-0ce74e49="" value="all">Alle</option>
                  <option data-v-0ce74e49="" value="in_progress">In Bearbeitung</option>
                  <option data-v-0ce74e49="" value="open">Offen</option>
                  <option data-v-0ce74e49="" value="completed">Abgeschlossen</option>
                </select></div>
            </div><!-- Issues Grid -->
            <div data-v-0ce74e49="" class="issues-grid">
              <div data-v-0ce74e49="" class="issue-card">
                <div data-v-0ce74e49="" class="issue-header"><span data-v-0ce74e49="" class="status-badge" style="background-color: rgb(147, 51, 234);">Offen</span></div><a data-v-0ce74e49="" href="https://github.com/example/issue1" target="_blank" class="issue-title-link">
                  <h3 data-v-0ce74e49="" class="issue-title">First issue</h3>
                </a>
                <div data-v-0ce74e49="" class="issue-description">
                  <p><strong>Bold text</strong> with <em>italic</em></p>
                </div>
                <div data-v-0ce74e49="" class="issue-footer">
                  <div data-v-0ce74e49="" class="labels"><span data-v-0ce74e49="" class="label" style="background-color: rgb(255, 0, 0);">bug</span></div><a data-v-0ce74e49="" href="https://github.com/example/issue1" target="_blank" class="details-link"> Details anzeigen → </a>
                </div>
              </div>
              <div data-v-0ce74e49="" class="issue-card">
                <div data-v-0ce74e49="" class="issue-header"><span data-v-0ce74e49="" class="status-badge" style="background-color: rgb(22, 163, 74);">Abgeschlossen</span></div><a data-v-0ce74e49="" href="https://github.com/example/issue2" target="_blank" class="issue-title-link">
                  <h3 data-v-0ce74e49="" class="issue-title">Second issue</h3>
                </a>
                <div data-v-0ce74e49="" class="issue-description">
                  <p>Simple issue description</p>
                </div>
                <div data-v-0ce74e49="" class="issue-footer">
                  <div data-v-0ce74e49="" class="labels"><span data-v-0ce74e49="" class="label" style="background-color: rgb(0, 255, 0);">enhancement</span></div><a data-v-0ce74e49="" href="https://github.com/example/issue2" target="_blank" class="details-link"> Details anzeigen → </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <!---->
    </div>
  </div>
</div>"
`;
